### PR TITLE
[React] Fix incorrect prefilling when creating new user

### LINKED ISSUE #8302: React: "Create a new user" dialog incorrectly prefills properties with data from a previously edited or detailed user
##### **Overview of the issue**

In the React frontend, under *User Management*, when clicking *Create a new user*, the dialog incorrectly prepopulates user property fields with data from the previously edited or previously in-detail-shown user - making creation of a new user impossible.

A workaround is to go back again (which causes `UserManagementUpdate` to dispatch a `RESET` action on `componentWillUnmount()`), then clicking  *Create a new user* again.

##### **Motivation for or Use Case**

Creating a user should always work.

##### **Reproduce the error**

* Visit vanilla React-based JHipster frontend. Log in as `admin`. Click *Administration* -> *User Management*.
* Either *view* details of an existing user, then click *Back* ... or *edit* an existing user and click *Save*.
* Now being back in the User management interface, click *Create a new user*.
* The *Create or edit a user* dialog is shown and the route is `/admin/user-management/new`. But data is prefilled with data from the previous user, making creation of a new user impossible - as also the ID property is prepopulated - which isn't allowed for new users.

##### **Related issues**
No.

##### **Suggest a Fix**
The problem is that the user edit dialog and user creation dialog both use the `UserManagementUpdate` component. This component uses the Redux state `{user} : UserManagementState`. This state is also handled by  `UserManagementDetail`.

##### **JHipster Version(s)**
generator-jhipster@5.3.1. I don't know if this is a regression.

##### **JHipster configuration**
```
<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "com.example",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "5.3.1",
    "applicationType": "monolith",
    "baseName": "tagadata",
    "packageName": "com.example",
    "packageFolder": "com/example",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": "spring-websocket",
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mysql",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "xxx",
    "clientFramework": "react",
    "useSass": true,
    "clientPackageManager": "npm",
    "testFrameworks": [
      "cucumber",
      "protractor"
    ],
    "jhiPrefix": "jhi",
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": [
      "en",
      "de"
    ]
  }
}
</pre>
</details>
```

##### **Browsers and Operating System**
Windows 7. Problem occurs at least in Chrome and Firefox, but problem is related to generated React codebase.

- [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by chrisdns**
This should fix #8302

- [ ] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [ ] Tests are added where necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

File: generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management-detail.tsx.ejs [hunk 1]
```diff
@@ -35,6 +35,10 @@ export class UserManagementDetail extends React.Component<IUserManagementDetailP
     this.props.getUser(this.props.match.params.login);
   }
 
+  componentWillUnmount() {
+    this.props.reset();
```
Comment (deepu105) [inline generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management-detail.tsx.ejs]: Actually the fix should be the other way around. The value needs to be reset during the did mount step of create. Can you try that way

Comment (chrisdns) [reply generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management-detail.tsx.ejs]: I tried it but this does not seem to work

Removing the ``user: action.payload.data`` in the reducer when ``CREATE`` or ``UPDATE`` actions are dispatched is working, or it is also possible to dispatch a reset just after the ``UPDATE`` without removing ``user: action.payload.data`` like the #8338 does, both work

Comment (deepu105) [reply generators/client/templates/react/src/main/webapp/app/modules/administration/user-management/user-management-detail.tsx.ejs]: ok lets do it in the reducer, the object can be reset on action success

**Comment by deepu105** [comment]
@chrisdns can you take a look at my comment

**Comment by deepu105** [comment]
closing in favor of https://github.com/jhipster/generator-jhipster/pull/8529
@chrisdns see my PR, the issue was with defaultValue

**Pull request closed without merging**